pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -366,3 +314,133 @@ func removeMonitorFromQuorum(context *clusterd.Context, clusterName, name string
 	logger.Infof("removed monitor %s", name)
 	return nil
 }
+
+func (c *Cluster) resolveInvalidMonitorPlacement(desiredMonCount int) (bool, error) {
+	nodeChoice, err := c.findInvalidMonitorPlacement(desiredMonCount)
+	if err != nil {
+		return true, fmt.Errorf("failed to find invalid mon placement %+v", err)
+	}
+
+	// no violation found
+	if nodeChoice == nil {
+		return false, nil
+	}
+
+	for name, nodeInfo := range c.mapping.Node {
+		if nodeInfo.Name == nodeChoice.Node.Name {
+			logger.Infof("rebalance: rescheduling mon %s from node %s", name, nodeInfo.Name)
+			c.failMon(len(c.clusterInfo.Monitors), desiredMonCount, name)
+			return true, nil
+		} else {
```

Comment:
FWIW my linter is complaining that `if block ends with a return statement, so drop this else and outdent its block`

Reply:
fixed